resource_transip_private_network_attachment.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/transip/gotransip/v6/repository"
+	"github.com/transip/gotransip/v6/vps"
+)
+
+func resourcePrivateNetworkAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePrivateNetworkAttachmentCreate,
+		Read:   resourcePrivateNetworkAttachmentRead,
+		// Update: resourcePrivateNetworkUpdate,
+		Delete: resourcePrivateNetworkAttachmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"private_network_id": {
+				Required: true,
+				Type:     schema.TypeString,
+				ForceNew: true,
+			},
+			"vps_id": {
+				Required: true,
+				Type:     schema.TypeString,
+				ForceNew: true,
+			},
+			"action": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourcePrivateNetworkAttachmentCreate(d *schema.ResourceData, m interface{}) error {
+	privateNetworkID := d.Get("private_network_id").(string)
+	vpsID := d.Get("vps_id").(string)
+
+	client := m.(repository.Client)
+	repository := vps.PrivateNetworkRepository{Client: client}
+
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		log.Printf("[DEBUG] terraform-provider-transip trying to attach network %s to vps %s \n", privateNetworkID, vpsID)
+
+		err := repository.AttachVps(vpsID, privateNetworkID)
+		if err != nil {
+			if strings.Contains(err.Error(), fmt.Sprintf("VPS '%s' has an action running, no modification is allowed", vpsID)) {
```

Comment:
Is there an other way we can deduct this is a temporary error from the API side? Like the HTTP return code being different? Comparing on the error string can be unreliable as it's not an API specification what text is returned.

Reply:
Valid point, it actually returns a 409. Unfortunately it returns a 409 for multiple causes:

1. The PrivateNetwork 'example-privatenetwork' is already locked to another action
2. VPS 'example-vps' has an action running, no modification is allowed
3. VPS 'example-vps' is customer locked, no modification is allowed
4. Actions on VPS 'example-vps' are temporary disabled

So it's still pretty hard to conclude if we should continue or not. What's your suggestion on this?

Reply:
In this case it not that bad I think as it would only result in a retryable error becoming a fatal error. On the next Terraform run the resource would be properly detected as created as well (if enough time has passed). So lets keep it in for now.